forwards_paper.tex

```diff
@@ -1167,43 +1167,56 @@ \section{Simplification algorithm}
 
 \end{taocpalg}
 
-The algorithm begins in \algref{S1} by allocating an empty list to store ancestral
+The algorithm begins in \algref{S0} by allocating an empty list to store ancestral
 segments for each of the $|\Nt_I|$ input nodes, and then creating the initial
 state for the samples in $S$. For each input sample with node ID $u$, we
 add a row in the output node table $\Nt_O$, and create a mapping for this
 new node $v$. After this initial state has been created, we then allocate our
 priority queue $Q$, and set the current input ID $u$ to $0$.
+Recall that the function $\taddrow{\Nt}{t}$ adds a new node to the node table $\Nt$
+with birth time $t$, and returns the ID (i.e., index) of that new node in $\Nt$.
+So, the operation $v \leftarrow \taddrow{\Nt_O}{\Nt_I[u]}$ looks up the birth time
+for input node ID $u$ in $\Nt_I$, adds a new node with this time to $\Nt_O$,
+and stores the resulting (output) node ID in $v$.
 
-The algorithm steps back up through the pedigree and steps \algref{S1}--\algref{S8}
+To step back up through the pedigree, steps \algref{S1}--\algref{S8}
```

Comment:
We're using "step" twice here for slightly different things. Might be confusing.